src/infinitempomatrix.jl

```diff
@@ -43,4 +44,91 @@ function InfiniteMPOMatrix(data::Vector{Matrix{ITensor}}, translator::Function)
   return InfiniteMPOMatrix(CelledVector(data, translator), 0, size(data)[1], false)
 end
 
-#nrange(H::InfiniteMPOMatrix) = [size(H[j])[1] - 1 for j in 1:nsites(H)]
+#
+#  Hm should have the form below.  Only link indices are shown.
+#
+#    I         0                     0           0      0
+#  M-->l=n     0                     0           0      0
+#    0    l=n-->M-->l=n-1 ...        0           0      0
+#    :         :                     :           :      :
+#    0         0          ...  l=2-->M-->l=1     0      0
+#    0         0          ...        0        l=1-->M   I
+#
+#  We need to capture the corner I's and the sub-diagonal Ms, and paste them together into an ITensor.
+#  This is facilutated by making all elements of Hm into order(4) tensors by adding dummy Dw=1 indices.
+#  We ITensors.directsum() to join all the blocks into the final ITensor.
+#  This code should be 100% dense/blocks-sparse agnostic.
+#
+function matrixITensorToITensor(Hm::Matrix{ITensor})::ITensor
```

Comment:
I would call this `cat_to_itensor(Hm::Matrix{ITensor})`, in analogy to Julia's block matrix concatenation interface:
```julia
julia> x = [[randn(2, 2)] [randn(2, 2)]; [randn(2, 2)] [randn(2, 2)]]
2×2 Matrix{Matrix{Float64}}:
 [-0.53919 -0.0631672; 1.70176 1.02657]  [-0.329444 1.71233; 1.68521 -0.113242]
 [0.0322956 0.240616; 2.24649 1.24855]   [-0.445636 1.02609; 1.72244 -0.10569]

julia> hvcat((2, 2), x...)
4×4 Matrix{Float64}:
 -0.53919   -0.0631672   0.0322956   0.240616
  1.70176    1.02657     2.24649     1.24855
 -0.329444   1.71233    -0.445636    1.02609
  1.68521   -0.113242    1.72244    -0.10569

julia> [x[1, 1] x[1, 2]; x[2, 1] x[2, 2]]
4×4 Matrix{Float64}:
 -0.53919    -0.0631672  -0.329444   1.71233
  1.70176     1.02657     1.68521   -0.113242
  0.0322956   0.240616   -0.445636   1.02609
  2.24649     1.24855     1.72244   -0.10569
```
(we can see this function as an ITensor version of that kind of concatenation).

Reply:
Done